x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_routing/state_management/stream_routing_state_machine/use_stream_routing.tsx

```diff
@@ -54,15 +55,19 @@ export const useStreamRoutingEvents = () => {
       editRule: (id: string) => {
         service.send({ type: 'routingRule.edit', id });
       },
-      forkStream: () => {
-        service.send({ type: 'routingRule.fork' });
+      forkStream: async (routingRule?: RoutingDefinition) => {
+        service.send({ type: 'routingRule.fork', routingRule });
+        await waitFor(service, (snapshot) => snapshot.matches({ ready: 'idle' }));
```

Comment:
so I've made this async so we can trigger an onSuccess event once completed but this only covers the happy path. if the forking fails then this will never resolve and just leave an unresolved promise right ? @tonyghiani is there a better suited approach for this ?